tasks/main.yml

```diff
@@ -16,15 +20,21 @@
 
 - name: Download Caddy
   get_url: url=https://caddyserver.com/download/build?os=linux&arch=amd64&features={{ caddy_features }} dest={{ caddy_home }}/caddy.tar.gz force=yes
-  when: caddy_releases_cache.changed or caddy_features_cache.changed
-  register: caddy_binary_cache
+  when: caddy_update and (caddy_releases_cache.changed or caddy_features_cache.changed)
+  register: caddy_downloaded_binary_cache
 
 - name: Download Caddy
   get_url: url=https://caddyserver.com/download/build?os=linux&arch=amd64&features={{ caddy_features }} dest={{ caddy_home }}/caddy.tar.gz
+  when: (not caddy_update) and (caddy_archive_file is not defined)
```

Comment:
Thanks to the assertion at line 8, you are correct. Again, I preferred being explicit as I understand this helps readability and maintainability.